### PR TITLE
Renamed StrictEqualityAssociativeArray to Map

### PR DESCRIPTION
**Pull request by aicioara**

### CODE REVIEW

**Comment by jtlan** [comment]
Not strictly the same as the ES6 map, but I think it's the best we can do unless we want to take a shimming library (the actual spec might also not be possible in ES5).

**Comment by jtlan** [comment]
Can we make this take generics?
```Typescript
Map<K, V>
```

**Comment by bluong** [comment]
Just in general if we want to rework a class or so, having `any` types in the final version is generally a bad smell.  Even for our general `data` arrays, we might actually type those things as `Object`s.

Be on the lookout for `any`s and get rid of them if you can.

**Comment by aicioara** [comment]
Solved. With this, I realized that our map is not actually a map. For example for `paddingException`, it should work both numbers and dates

File: src/core/dataset.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ module Plottable {
   export class Dataset extends Core.PlottableObject {
     private _data: any[];
     private _metadata: any;
-    private _accessor2cachedExtent: Utils.StrictEqualityAssociativeArray;
+    private _accessor2cachedExtent: Utils.Map<any, any>;
```
Comment (jtlan) [inline src/core/dataset.ts]: Actually, this variable is no longer used.

File: src/core/domainer.ts [hunk 2]
```diff
@@ -6,8 +6,8 @@ module Plottable {
     private _doNice = false;
     private _niceCount: number;
     private _padProportion = 0.0;
-    private _paddingExceptions = new Utils.StrictEqualityAssociativeArray();
-    private _includedValues = new Utils.StrictEqualityAssociativeArray();
+    private _paddingExceptions: Utils.Map<any, any>;
+    private _includedValues: Utils.Map<any, any>;
```
Comment (jtlan) [inline src/core/domainer.ts]: `Domainer` as a whole probably needs to take a generic (the `<D>` on a `Scale<D, R>`), but we can put that aside for now.

**Pull request merged**